libs/ledger-live-common/src/deviceSDK/commands/firmwareUpdate/installFirmware.ts

```diff
@@ -0,0 +1,148 @@
+import { Observable, of, throwError } from "rxjs";
+import URL from "url";
+import Transport, { TransportStatusError } from "@ledgerhq/hw-transport";
+import type { FinalFirmware, OsuFirmware } from "@ledgerhq/types-live";
+import type { DeviceInfo, SocketEvent } from "@ledgerhq/types-live";
+import { version as livecommonversion } from "../../../../package.json";
+import { getEnv } from "../../../env";
+//import { createMockSocket, bulkSocketMock, secureChannelMock } from "../../api/socket.mock";
+import { log } from "@ledgerhq/logs";
+import { createDeviceSocket } from "../../../api/socket";
+import { catchError, filter, map } from "rxjs/operators";
+import {
+  ManagerFirmwareNotEnoughSpaceError,
+  UserRefusedFirmwareUpdate,
+  DeviceOnDashboardExpected,
+  ManagerDeviceLockedError,
+} from "@ledgerhq/errors";
+import { UnresponsiveCmdEvent } from "../core";
+
+export type InstallFirmwareCommandRequest = {
+  targetId: DeviceInfo["targetId"];
+  firmware: OsuFirmware | FinalFirmware;
+};
+
+type FilteredSocketEvent =
+  | {
+      type: "bulk-progress";
+      progress: number;
+      index: number;
+      total: number;
+    }
+  | {
+      type: "device-permission-requested";
+      wording: string;
+    };
+
+export type InstallFirmwareCommandEvent =
+  | {
+      type: "progress";
+      progress: number;
+    }
+  | {
+      type: "allowManagerRequested";
+    }
+  | {
+      type: "firmwareInstallPermissionRequested";
+    }
+  | {
+      type: "firmwareInstallPermissionGranted";
+    }
+  | UnresponsiveCmdEvent;
+
+export function installFirmwareCommand(
```

Comment:
We should start documenting what each command (and task) are doing - because just by reading the name, i don't see which one does what between `installFirmware` and `flashMcuOrBootloader`

Reply:
The biggest difference is the endpoint to which we create the secure channel socket connection.

In installFirmware it's the `/install` one which is also the same that's used for installing apps. In flashMcuOrBootloader it's the `/mcu` one.

There's also some extra handling of different types of events in the `/install` since there are more steps or types of apdu that can come from that endpoint.

We should probably document all the endpoints as well that we have. Which ones require secure connections with the HSM etc